PocketKit/Sources/Sync/PocketSource.swift

```diff
@@ -649,15 +649,17 @@ extension PocketSource {
     }
 
     public func fetchOrCreateSavedItem(with remoteID: String, and remoteParts: SavedItem.RemoteSavedItem?) -> SavedItem? {
+        let savedItem = (try? space.fetchSavedItem(byRemoteID: remoteID))
+
         guard let remoteParts, let url = URL(string: remoteParts.url) else {
-            Log.breadcrumb(category: "sync", level: .warning, message: "Skipping updating of SavedItem because we do not have a valid url or we have no remoteParts")
-            return nil
+            Log.breadcrumb(category: "sync", level: .debug, message: "SavedItem found and don't need to create one")
```

Comment:
@cyndichin isnt the meaning of the guard here, that we have an invalid URL? not that we don't need to create one.

Reply:
This function purpose is to check if there's a saved item already existing, otherwise we have to create it from the remote parts. If we want to check if a url exists, then would checking on the `savedItem.url` be better? A local savedItem would not have any `remoteParts`.